packages/contracts-bedrock/src/governance/ProposalValidator.sol

```diff
@@ -501,53 +541,62 @@ contract ProposalValidator is OwnableUpgradeable, ReinitializableBase, ISemver {
+        IProposalTypesConfigurator.ProposalType memory proposalTypeConfig = IProposalTypesConfigurator(
+            GOVERNOR.PROPOSAL_TYPES_CONFIGURATOR()
+        ).proposalTypes(proposalTypesData[_proposalType].idInConfigurator);
+        address votingModule = proposalTypeConfig.module;
+
+        // Validate voting module exists
+        if (bytes(proposalTypeConfig.name).length == 0) {
+            revert ProposalValidator_InvalidVotingModule();
+        }
 
-        // Generate unique proposal hash
-        proposalHash_ = _hashProposalWithModule(votingModule, proposalVotingModuleData, keccak256(bytes(_description)));
+        // Generate unique proposal ID
+        proposalId_ = _hashProposalWithModule(votingModule, proposalVotingModuleData, keccak256(bytes(_description)));
 
-        ProposalData storage proposal = _proposals[proposalHash_];
+        ProposalData storage proposal = _proposals[proposalId_];
 
-        // Prevent duplicate proposals with same hash
+        // Prevent duplicate proposals with same ID
         if (proposal.proposer != address(0)) {
             revert ProposalValidator_ProposalAlreadySubmitted();
         }
 
         // Check if proposal already exists in OptimismGovernor
-        if (GOVERNOR.proposalSnapshot(uint256(proposalHash_)) != 0) {
+        if (GOVERNOR.proposalSnapshot(proposalId_) != 0) {
             revert ProposalValidator_ProposalAlreadySubmitted();
         }
 
         // Store proposal metadata
-        proposal.proposer = msg.sender;
+        proposal.proposer = _msgSender();
         proposal.proposalType = _proposalType;
         proposal.votingCycle = _votingCycle;
 
-        emit ProposalSubmitted(proposalHash_, msg.sender, _description, _proposalType);
-        emit ProposalVotingModuleData(proposalHash_, proposalVotingModuleData);
+        emit ProposalSubmitted(proposalId_, _msgSender(), _description, _proposalType);
+        emit ProposalVotingModuleData(proposalId_, proposalVotingModuleData);
     }
 
     /// @notice Approves a proposal before being moved for voting.
     /// @dev This function should only be called by the top delegates.
-    /// @param _proposalHash The hash of the proposal to approve
+    /// @param _proposalId The ID of the proposal to approve
     /// @param _attestationUid The UID of the attestation for the delegate to approve the proposal
-    function approveProposal(bytes32 _proposalHash, bytes32 _attestationUid) external {
+    function approveProposal(uint256 _proposalId, bytes32 _attestationUid) external {
         address _delegate = _msgSender();
-        ProposalData storage proposal = _proposals[_proposalHash];
+        ProposalData storage proposal = _proposals[_proposalId];
         // check if the proposal exists
-        if (proposal.proposer == address(0)) {
+        // proposal.votingCycle should never be 0, voting cycles already exist before the ProposalValidator is deployed
```

Comment:
Note here that when this is deployed the previous voting cycle should not be left without data, as otherwise this function will fail. Similarly if it's possible to have votingCycle one, the same will happen